larreco/HitFinder/HitFinderTools/PeakFitterGaussian_tool.cc

```diff
@@ -270,13 +268,222 @@ namespace reco_tool {
+
+      diff *= diff;
+
+      // We want to avoid adding new Gaussians in the tails
+      float peakDist(std::numeric_limits<float>::max());
+      for (auto const& hitCandidate : hitCandidateVec)
+        peakDist = std::min(peakDist, std::abs(hitCandidate.hitCenter - float(startTime) - i));
+
+      // Or too close to a peak
+      if (peakDist < 3) continue;
+
+      diff *= std::log(peakDist);
+
+      if (std::abs(diff) > std::abs(maxDiff)) {
+        maxDiff = diff;
+        maxDiffPos = i;
+      }
+    }
+
+    if (maxDiffPos == std::numeric_limits<int>::max())
+      return ICandidateHitFinder::HitCandidate{0,
+                                               0,
+                                               0,
+                                               0,
+                                               0,
+                                               0,
+                                               std::numeric_limits<float>::lowest(),
+                                               std::numeric_limits<float>::lowest(),
+                                               std::numeric_limits<float>::lowest()};
+
+    // Recover the actual diff
+    maxDiff = (waveform[startTime + maxDiffPos] - fittedGaus.Eval(maxDiffPos));
+
+    int lowLim(maxDiffPos);
+    int highLim(maxDiffPos);
+    const bool useMax(maxDiff > 0);
+
+    // Find the point where the excess crosses the fitted Gaussian
+    if (useMax) {
+      while (lowLim > 0 && waveform[startTime + lowLim] > fittedGaus.Eval(lowLim))
+        lowLim--;
+
+      while (highLim < roiSize && waveform[startTime + highLim] > fittedGaus.Eval(highLim))
+        highLim++;
+    }
+    else {
+      while (lowLim > 0 && waveform[startTime + lowLim] < fittedGaus.Eval(lowLim))
+        lowLim--;
+
+      while (highLim < roiSize && waveform[startTime + highLim] < fittedGaus.Eval(highLim))
+        highLim++;
+    }
+
+    const float amplitude(std::max(std::abs(2 * maxDiff), waveform[startTime + maxDiffPos] / 2.f));
+
+    return ICandidateHitFinder::HitCandidate{
+      0, 0, 0, 0, 0, 0, float(maxDiffPos + startTime), 0.5f * (highLim - lowLim), amplitude};
+  }
+
+  std::pair<ICandidateHitFinder::HitCandidate, IPeakFitter::PeakFitParams_t>
+  PeakFitterGaussian::FindShiftedGaussian(
+    const ICandidateHitFinder::HitCandidateVec& hitCandidateVec,
+    const PeakParamsVec& fittedPeakVec) const
+  {
+    float minDiff(std::numeric_limits<float>::max());
+    ICandidateHitFinder::HitCandidate const* minHit;
+    PeakFitParams_t const* minPeak;
+
+    for (auto const& hitCand : hitCandidateVec) {
+      for (auto const& fittedPeak : fittedPeakVec) {
+        const float offset(hitCand.hitCenter - fittedPeak.peakCenter);
+        if (std::abs(offset) < minDiff) {
+          minDiff = offset;
+          minHit = &hitCand;
+          minPeak = &fittedPeak;
+        }
+      }
+    }
+    return std::pair<ICandidateHitFinder::HitCandidate, PeakFitParams_t>(*minHit, *minPeak);
```

Comment:
```suggestion
    return {*minHit, *minPeak};
```

Reply:
As a bonafide old fogey (I get letters from AARP all the time) I always preferred typing out the full qualifier so that when I come back to read the code a few epochs later I can more easily remember what is going on. I get that the brackets are "easier" both to type and likely also in the event of changes to the definition. But throw the old dog a bone here and leave in the extra typing?

Reply:
Well...the explicit return type is stated a few lines above at the beginning of the function definition (L. 466).  But I don't insist on the curly braces.  Would you feel better with:

```C++
return std::make_pair(*minHit, *minPeak);
```

or is that still too much obfuscation?